src/monster.cpp

```diff
@@ -1740,6 +1740,10 @@ void monster::process_effects()
         add_msg( m_warning, _( "The %s seems a little healthier." ), name().c_str() );
     }
 
+    if( has_flag( MF_REGENERATES_IN_DARK ) && heal( 40 - int( g->m.ambient_light_at( pos() ) ) ) > 0 && one_in( 2 ) && g->u.sees( *this ) ) {
+        add_msg( m_warning, _( "The %s uses the darkness to regenerate." ), name().c_str() );
```

Comment:
I'd avoid specifying that it "uses darkness". The flag may be useful for other monsters that regenerate for a different reason.

How about the same generic `_( "The %s is visibly regenerating!" )` as above? That'd even save a line for translators.